idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/conversion/IconPackConversionState.kt

```diff
@@ -1,23 +1,26 @@
 package io.github.composegears.valkyrie.ui.screen.mode.iconpack.conversion
 
-import androidx.compose.ui.graphics.painter.Painter
 import java.nio.file.Path
 
 sealed interface IconPackConversionState {
 
     data object IconsPickering : IconPackConversionState
 
-    data class BatchFilesProcessing(
-        val iconsToProcess: List<BatchIcon> = emptyList(),
-    ) : IconPackConversionState {
-
-        val exportEnabled: Boolean
-            get() = iconsToProcess.isNotEmpty() &&
-                iconsToProcess.all { it is BatchIcon.Valid } &&
-                iconsToProcess.all { icon ->
-                    icon.iconName.value.isNotEmpty() &&
-                        !icon.iconName.value.contains(" ")
-                }
+    sealed interface BatchProcessing : IconPackConversionState {
+
+        data class IconPackCreationState(val icons: List<BatchIcon>) : BatchProcessing {
+
+            val exportEnabled: Boolean
+                get() = icons.isNotEmpty() &&
+                    icons.all { it is BatchIcon.Valid } &&
+                    icons.all { icon ->
+                        icon.iconName.value.isNotEmpty() &&
+                            !icon.iconName.value.contains(" ")
+                    }
+        }
```

Comment:
We should not leave complex logic into states, seems just need `exportEnabled` in `IconPackCreationState`, these check logic should be placed in ViewModel or Presenter.

Reply:
State classes should look as simple as possible, avoiding containing complex logic.

Reply:
yes, in case processing huge amount of icons - potential bottleneck

Reply:
I'll take a look later.

Reply:
I changed in scope of this ticket, will update branch soon

Reply:
https://github.com/ComposeGears/Valkyrie/compare/1e3de3c0e7e48cb4daa2f91ed91c668a26b4ebd6..e4661726e784e42e9a5b98b71269edccb36e6cd3